superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts

```diff
@@ -27,10 +33,29 @@ import { PivotTableQueryFormData } from '../types';
 export default function buildQuery(formData: PivotTableQueryFormData) {
   const { groupbyColumns = [], groupbyRows = [] } = formData;
   // TODO: add deduping of AdhocColumns
-  const groupbySet = new Set([
-    ...ensureIsArray<QueryFormColumn>(groupbyColumns),
-    ...ensureIsArray<QueryFormColumn>(groupbyRows),
-  ]);
+  const groupbySet = Array.from(
+    new Set([
+      ...ensureIsArray<QueryFormColumn>(groupbyColumns),
+      ...ensureIsArray<QueryFormColumn>(groupbyRows),
+    ]),
+  ).map(col => {
+    if (
+      isPhysicalColumn(col) &&
+      formData.time_grain_sqla &&
+      isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) &&
+      formData?.datetimeColumnLookup[col]
```

Comment:
nit: do we need a null check here (we don't do null check in line 44). Maybe we should do `formData.datetimeColumnLookup?.[col]` instead?

Reply:
done.